File: backend/user-auth-service/src/main/java/com/tripify/user_auth_service/dto/request/PaymentMethodDto.java
package com.tripify.user_auth_service.dto.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PaymentMethodDto {
    private UUID id;
    private String cardProvider;
    private String cardNumber;
    private String lastFourDigits;
    private String expirationMonthYear;
}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/PaymentMethodDto.kt
package com.tripify.tripify_android.data.model

data class PaymentMethodDto(
    val id: String? = null,
    val cardProvider: String,
    val cardNumber: String? = null,
    val lastFourDigits: String? = null,
    val expirationMonthYear: String,
    val defaultCard: Boolean = false
) {
    override fun toString(): String =
        "PaymentMethodDto(id=$id, cardProvider=$cardProvider, cardNumber=${if (cardNumber == null) "null" else "***"}, " +
            "lastFourDigits=$lastFourDigits, expirationMonthYear=$expirationMonthYear, defaultCard=$defaultCard)"
}
